File: File and Image Utilities/merge.py
import os
import shutil

# Source folders
folder1 = r'/home/homam/Desktop/Projects/jupyter-for-tumor-ai/first ai test/data/no tumor/rename no tumor/'
folder2 = r'/home/homam/Desktop/Projects/jupyter-for-tumor-ai/first ai test/data/no tumor/rename no tumor 3/'
folder3 = r'/home/homam/Desktop/Projects/jupyter-for-tumor-ai/first ai test/data/no tumor/rename no tumor part_1/'
folder4 = r'/home/homam/Desktop/Projects/jupyter-for-tumor-ai/first ai test/data/no tumor/rename no tumor part_2/'
folder5 = r'/home/homam/Desktop/Projects/jupyter-for-tumor-ai/first ai test/data/no tumor/rename no tumor part_3/'
folder6 = r'/home/homam/Desktop/Projects/jupyter-for-tumor-ai/first ai test/data/no tumor/rename no tumor part_4/'

# Destination folder
destination_folder = r'/home/homam/Desktop/Projects/jupyter-for-tumor-ai/first ai test/data/no tumor/'

# Create the destination folder if it doesn't exist
os.makedirs(destination_folder, exist_ok=True)

# Copy and rename images from folder1
for i, filename in enumerate(os.listdir(folder1)):
    src_path = os.path.join(folder1, filename)
    dest_path = os.path.join(destination_folder, f'{i + 1}.jpg')
    shutil.copy(src_path, dest_path)

# Copy and rename images from folder2
for i, filename in enumerate(os.listdir(folder2)):
    src_path = os.path.join(folder2, filename)
    dest_path = os.path.join(destination_folder, f'{i + 1 + len(os.listdir(folder1))}.jpg')
    shutil.copy(src_path, dest_path)

# Copy and rename images from folder3
for i, filename in enumerate(os.listdir(folder3)):
    src_path = os.path.join(folder3, filename)
    dest_path = os.path.join(destination_folder, f'{i + 1 + len(os.listdir(folder1)) + len(os.listdir(folder2))}.jpg')
    shutil.copy(src_path, dest_path)
    
# Copy and rename images from folder3
for i, filename in enumerate(os.listdir(folder4)):
    src_path = os.path.join(folder4, filename)
    dest_path = os.path.join(destination_folder, f'{i + 1 + len(os.listdir(folder1)) + len(os.listdir(folder2))}.jpg')
    shutil.copy(src_path, dest_path)

# Copy and rename images from folder3
for i, filename in enumerate(os.listdir(folder5)):
    src_path = os.path.join(folder5, filename)
    dest_path = os.path.join(destination_folder, f'{i + 1 + len(os.listdir(folder1)) + len(os.listdir(folder2))}.jpg')
    shutil.copy(src_path, dest_path)

# Copy and rename images from folder3
for i, filename in enumerate(os.listdir(folder6)):
    src_path = os.path.join(folder6, filename)
    dest_path = os.path.join(destination_folder, f'{i + 1 + len(os.listdir(folder1)) + len(os.listdir(folder2))}.jpg')
    shutil.copy(src_path, dest_path)


print("Images merged and renamed successfully.")

File: File and Image Utilities/merge.cpp
#include <iostream>
#include <filesystem>
#include <string>

namespace fs = std::filesystem;

void copyAndRenameImages(const std::vector<std::string>& sourceFolders, const std::string& destinationFolder) {
    // Create the destination folder if it doesn't exist
    fs::create_directories(destinationFolder);

    int globalIndex = 1; // Starting index for renaming files

    // Iterate through each source folder
    for (const auto& folder : sourceFolders) {
        // Iterate through all files in the current source folder
        for (const auto& entry : fs::directory_iterator(folder)) {
            if (entry.is_regular_file()) {
                std::string fileName = entry.path().filename().string();
                std::string destinationPath = destinationFolder + "/" + std::to_string(globalIndex++) + ".jpg";
                
                // Copy the file to the destination folder with the new name
                fs::copy(entry.path(), destinationPath, fs::copy_options::overwrite_existing);
            }
        }
    }

    std::cout << "Images merged and renamed successfully." << std::endl;
}

int main() {
    // List of source folders
    std::vector<std::string> sourceFolders = {
        "/home/homam/Desktop/Projects/jupyter-for-tumor-ai/first ai test/data/no tumor/rename no tumor/",
        "/home/homam/Desktop/Projects/jupyter-for-tumor-ai/first ai test/data/no tumor/rename no tumor 3/",
        "/home/homam/Desktop/Projects/jupyter-for-tumor-ai/first ai test/data/no tumor/rename no tumor part_1/",
        "/home/homam/Desktop/Projects/jupyter-for-tumor-ai/first ai test/data/no tumor/rename no tumor part_2/",
        "/home/homam/Desktop/Projects/jupyter-for-tumor-ai/first ai test/data/no tumor/rename no tumor part_3/",
        "/home/homam/Desktop/Projects/jupyter-for-tumor-ai/first ai test/data/no tumor/rename no tumor part_4/"
    };

    // Destination folder
    std::string destinationFolder = "/home/homam/Desktop/Projects/jupyter-for-tumor-ai/first ai test/data/no tumor/";

    // Call the function to copy and rename images
    copyAndRenameImages(sourceFolders, destinationFolder);

    return 0;
}
